FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/AzureBlobJobStore.cs

```diff
@@ -131,51 +197,24 @@ public async Task<IEnumerable<Job>> GetActiveJobsAsync(CancellationToken cancell
             return jobs;
         }
 
-        public async Task UpdateJobAsync(Job job, CancellationToken cancellationToken = default)
-        {
-            EnsureArg.IsNotNull(job, nameof(job));
-
-            job.LastHeartBeat = DateTimeOffset.UtcNow;
-
-            try
-            {
-                await _blobContainerClient.UpdateBlobAsync(
-                    $"{AzureBlobJobConstants.ActiveJobFolder}/{job.Id}.json",
-                    job.ToStream(),
-                    cancellationToken: cancellationToken);
-
-                _logger.LogInformation("Update job '{job.Id}' successfully.", job.Id);
-            }
-            catch (Exception ex)
-            {
-                _logger.LogError(ex, "Failed to update job '{job.Id}'", job.Id);
-                throw;
-            }
-        }
-
-        public async Task CompleteJobAsync(Job job, CancellationToken cancellationToken = default)
+        private async Task CompleteJobAsyncInternal(Job job, bool releaseLock, CancellationToken cancellationToken)
```

Comment:
When will we release the lock and when will not?

Reply:
If we complete a job when acquiring job, we will not release lock to continue execute. If we complete a job when we finished executing the job, the lock is released.